webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSDependenciesProcessor.java

```diff
@@ -66,8 +67,11 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
         for(ModuleIdentifier api : JAVAEE_APIS) {
             moduleSpec.addSystemDependency(new ModuleDependency(moduleLoader, api, false, false, true, false));
         }
-        //After jboss modules 2.0, the xercers module is not added
-        moduleSpec.addSystemDependency(new ModuleDependency(moduleLoader, XERCES_IMPL, true, false, true, false));
 
+        //After jboss modules 2.0, the xerces module is not added and this is required for jaxb 2.3.x(EE8) to pass the TCK
+        //But this is not needed for jaxb 3.x in WildFly Preview(EE10)
+        if (Endpoint.class.getPackage().getImplementationVersion().startsWith("3")) {
+            moduleSpec.addSystemDependency(new ModuleDependency(moduleLoader, XERCES_IMPL, true, false, true, false));
+        }
```

Comment:
I almost wonder if we should just remove this. Once we merge the big-bang change, 2.3.x won't be used anyway.

Reply:
@jamezp Will there be any branch/tag before the big-bang change ?

Reply:
@jimma We will create a tag before merging the big-bang branch.

Reply:
Ah, I say we keep it then so we can get a good tag before we remove it after the big-bang.